### DISCUSSION TITLE
How to call only one time a handler event ("app.on_event") using more then one worker

### DISCUSSION THREAD

**Discussion by mateusjs** [fastapi/fastapi#6318] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi_utils.tasks import repeat_every
from fastapi import FastAPI

app = FastAPI()

@app.on_event("startup")
@repeat_every(seconds=5, wait_first=True)
def every_five_seconds():
    print("5 seconds")
```

### Description

When i start my application with:

`uvicorn main:app --workers 4`

My application is calling the handler "startup" for each worker, so the "every_five_seconds" method, is called four times in a row each five seconds.

It's possible to call it once each five seconds?

### Operating System

Windows

### Operating System Details

Windows 11 Version 21H2 22000.434

### FastAPI Version

0.63.0

### Python Version

3.9.7

**Comment by panla** [comment]
Singleton mode?

**Comment by Dustyposa** [comment]
You can use gunicorn on start `OP`.
[reference](https://stackoverflow.com/questions/24101724/gunicorn-with-multiple-workers-is-there-an-easy-way-to-execute-certain-code-onl)
Or use a way make sure one worker work, eg: add a unique lock for start.

**Comment by mateusjs** [comment]
> You can use gunicorn on start `OP`. [reference](https://stackoverflow.com/questions/24101724/gunicorn-with-multiple-workers-is-there-an-easy-way-to-execute-certain-code-onl) Or use a way make sure one worker work, eg: add a unique lock for start.

Thanks for the help, it worked for only one taks, so when the application is stopped i can make something and only do it one time.

But my problem right now is more complicated. 

I'm using @repeat_every to create a task and validate an external api status, and this task should be done each 10 minutes, but i can't figure it out how to do it using server hook.

**Comment by mateusjs** [comment]
Already answer -> https://github.com/tiangolo/fastapi/issues/2425#issuecomment-734790381

This is the same question, so i will close the issue.
